core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1326,6 +1355,56 @@ int leaderEpoch() {
         return leaderEpoch;
     }
 
+    /**
+     * Records the fetch lock ratio metric. The metric is the ratio of the time the fetch lock was
+     * acquired to the total time since the last lock acquisition. The total time is calculated by
+     * adding the time since the last lock acquisition to the time the fetch lock was acquired.
+     *
+     * @param acquiredTime The time duration the fetch lock was acquired.
+     */
+    // Visible for testing
+    void recordFetchLockRatioMetric(long acquiredTime) {
```

Comment:
I think there's still some confused naming here. Some of these "times" are the current time, and some are durations. Please make the naming consistent so it's easier to follow.

Reply:
I have updated the code with variable names as `durationMs`. Also I changed the way idle time is calculated by taking an explicit variable, where the code becomes easy to understand. Please let me know your thoughts.